tests/unit/parser/src/test_authenticate_manifest.c

```diff
@@ -0,0 +1,736 @@
+			0x84, /* array (4 elements) */
+			0x43, /* protected: bytes(3) / serialized map */
+				0xa0, /* empty header_map  */
+			0xa0, /* unprotected: header_map (0 elements) */
+			0xf6, /* payload: nil */
+			0x58, 0x40, /* bytes(64) : signature */
+			0xe3, 0x50, 0x5f, 0x7a, 0xb7, 0x0b, 0xd3, 0xa0,
+			0xe0, 0x49, 0x16, 0xf3, 0x7b, 0x0d, 0x72, 0x51,
+			0xaa, 0x6f, 0x52, 0xca, 0x12, 0xc7, 0xed, 0xaa,
+			0x88, 0x6a, 0x41, 0x29, 0xa2, 0x98, 0xca, 0x6a,
+			0x1e, 0xcc, 0x2a, 0x57, 0x95, 0x5c, 0x6b, 0xf4,
+			0xcc, 0xb9, 0xf0, 0x1d, 0x68, 0x4d, 0x5d, 0x1c,
+			0x47, 0x74, 0xdf, 0xfb, 0xe5, 0x08, 0xa0, 0x34,
+			0x43, 0x1f, 0xea, 0xfa, 0x60, 0x84, 0x8a, 0x2c,
+
+	0x03, /* suit-manifest */
+	0x48, /* bytes(8) */
+		'M', 'a', 'n', 'i', 'f', 'e', 's', 't',
+};
+
+static uint8_t cbor_envelope_signed_empty_header_map[] = {
+	0xd8, 0x6b, /* tag(107) : SUIT_Envelope */
+	0xa2, /* map (2 elements) */
+
+	0x02, /* suit-authentication-wrapper */
+		0x58, 0x4c, /* bytes(76) */
+		0x82, /* array (2 elements) */
+			0x41, /* bytes(1) */
+			0x09,
+
+			0x58, 0x47, /* bytes(71): SUIT_Authentication_Block */
+			0xd2, /* tag(18) : COSE_Sign1 */
+			0x84, /* array (4 elements) */
+			0x40, /* protected: bytes(0) / serialized map */
+			0xa0, /* unprotected: header_map (0 elements) */
+			0xf6, /* payload: nil */
+			0x58, 0x40, /* bytes(64) : signature */
+			0xe3, 0x50, 0x5f, 0x7a, 0xb7, 0x0b, 0xd3, 0xa0,
+			0xe0, 0x49, 0x16, 0xf3, 0x7b, 0x0d, 0x72, 0x51,
+			0xaa, 0x6f, 0x52, 0xca, 0x12, 0xc7, 0xed, 0xaa,
+			0x88, 0x6a, 0x41, 0x29, 0xa2, 0x98, 0xca, 0x6a,
+			0x1e, 0xcc, 0x2a, 0x57, 0x95, 0x5c, 0x6b, 0xf4,
+			0xcc, 0xb9, 0xf0, 0x1d, 0x68, 0x4d, 0x5d, 0x1c,
+			0x47, 0x74, 0xdf, 0xfb, 0xe5, 0x08, 0xa0, 0x34,
+			0x43, 0x1f, 0xea, 0xfa, 0x60, 0x84, 0x8a, 0x2c,
+
+	0x03, /* suit-manifest */
+	0x48, /* bytes(8) */
+		'M', 'a', 'n', 'i', 'f', 'e', 's', 't',
+};
+
+static uint8_t cbor_envelope_signed_no_protected[] = {
+	0xd8, 0x6b, /* tag(107) : SUIT_Envelope */
+	0xa2, /* map (2 elements) */
+
+	0x02, /* suit-authentication-wrapper */
+		0x58, 0x4b, /* bytes(75) */
+		0x82, /* array (2 elements) */
+			0x41, /* bytes(1) */
+			0x09,
+
+			0x58, 0x46, /* bytes(70): SUIT_Authentication_Block */
+			0xd2, /* tag(18) : COSE_Sign1 */
+			0x83, /* array (3 elements) */
+			0xa0, /* unprotected: header_map (0 elements) */
+			0xf6, /* payload: nil */
+			0x58, 0x40, /* bytes(64) : signature */
+			0xe3, 0x50, 0x5f, 0x7a, 0xb7, 0x0b, 0xd3, 0xa0,
+			0xe0, 0x49, 0x16, 0xf3, 0x7b, 0x0d, 0x72, 0x51,
+			0xaa, 0x6f, 0x52, 0xca, 0x12, 0xc7, 0xed, 0xaa,
+			0x88, 0x6a, 0x41, 0x29, 0xa2, 0x98, 0xca, 0x6a,
+			0x1e, 0xcc, 0x2a, 0x57, 0x95, 0x5c, 0x6b, 0xf4,
+			0xcc, 0xb9, 0xf0, 0x1d, 0x68, 0x4d, 0x5d, 0x1c,
+			0x47, 0x74, 0xdf, 0xfb, 0xe5, 0x08, 0xa0, 0x34,
+			0x43, 0x1f, 0xea, 0xfa, 0x60, 0x84, 0x8a, 0x2c,
+
+	0x03, /* suit-manifest */
+	0x48, /* bytes(8) */
+		'M', 'a', 'n', 'i', 'f', 'e', 's', 't',
+};
+
+static uint8_t cbor_envelope_signed_invalid_alg_id[] = {
```

Comment:
My first impression looking in this code and test name was that there will be unsupported alg_id used but there is different situation, we've got supported alg_id and wrong signature? In meaning that signature is calculated using ES256 but alg_id is set ot ES512 - is that correct? If yes then consider renaming this variable to ...invalid_signature

Reply:
This test is meant to detect the unsupported by the CDDL `alg_id`.
The signature value is not verified in unit tests (this is platform level check).
Indeed, the signature value should be longer for ES512, but if we do so, than many other places may recognize this situation as faulty (too long structure, not a 64-byte long signature), so in fact this test verifies that the ZCBOR parser does not allow other values for the `alg_id` field - since it is the only value allowed, ZCBOR does not even inform decoder code about the algorithm type.